src/parser/jobs/rpr/modules/HarvestMoon.tsx

```diff
@@ -0,0 +1,90 @@
+import {Trans} from '@lingui/react'
+import {DataLink} from 'components/ui/DbLink'
+import {Event} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import {Data} from 'parser/core/modules/Data'
+import {Invulnerability} from 'parser/core/modules/Invulnerability'
+import {UnableToAct} from 'parser/core/modules/UnableToAct'
+import React from 'react'
+
+const SOULSOW_BUFFER = 1000
+
+export class HarvestMoon extends Analyser {
+	static override handle = 'harvestMoon'
+
+	@dependency private checklist!: Checklist
+	@dependency private data!: Data
+	@dependency private invulnerability!: Invulnerability
+	@dependency private unableToAct!: UnableToAct
+
+	private harvestMoonCasts = 0
+
+	override initialise() {
+		super.initialise()
+
+		this.addEventHook(
+			filter<Event>()
+				.source(this.parser.actor.id)
+				.type('action')
+				.action(this.data.actions.HARVEST_MOON.id),
+			() => this.harvestMoonCasts++)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private windowFilterCallback(inputWindow: {start: number, end: number}): boolean {
+		const unableToActWindow = this.unableToAct.getWindows(inputWindow)[0]
+
+		if (unableToActWindow == null) {
+			return false
+		}
+
+		if (unableToActWindow.start - inputWindow.start >= this.data.actions.SOULSOW.castTime + SOULSOW_BUFFER) {
+			return true
+		}
+
+		return this.windowFilterCallback({start: unableToActWindow.end, end: inputWindow.end})
+	}
+
+	private getExpectedUses(): number {
+		const ADJUSTED_CAST = this.data.actions.SOULSOW.castTime + SOULSOW_BUFFER
+		const invulnWindows = this.invulnerability.getWindows().filter((w) => w.end - w.start >=  ADJUSTED_CAST)
+
+		if (this.unableToAct.getDuration() > 0) {
+			return invulnWindows.filter(this.windowFilterCallback).length + 1
+		}
+
+		return invulnWindows.length + 1
+	}
+
+	private getUsedPercentage(expectedUses: number): string {
+		return (this.harvestMoonCasts / expectedUses * 100).toFixed(2)
+	}
+
+	private onComplete() {
+		const expectedUses = this.getExpectedUses()
+		const hmUsesPercent = this.getUsedPercentage(expectedUses)
```

Comment:
Expected uses is only used here, could probably change the function to just return the used percent for you outright.

Reply:
the value is used on line 83 as well, for the display

Reply:
Ah you're right, my bad then.